website/www/site/content/en/documentation/runtime/environments.md

```diff
@@ -17,147 +17,232 @@ limitations under the License.
-3. [Reimage](#building-container-images) the container.
 
-### Testing customized images
+**NOTE**: After pushing a container image, you should verify the remote image ID and digest should match the local image ID and digest, output from `docker build` or `docker images`.
 
-To test a customized image locally, run a pipeline with PortableRunner and set the `--environment_config` flag to the image path:
+#### Modifying a source Dockerfile in Beam {#modifying-dockerfiles}
 
-{{< highlight class="runner-direct" >}}
-python -m apache_beam.examples.wordcount \
---input=/path/to/inputfile \
---output /path/to/write/counts \
---runner=PortableRunner \
---job_endpoint=embed \
---environment_config=path/to/container/image
-{{< /highlight >}}
+This method will require building image artifacts from Beam source. For additional instructions on setting up your development environment, see the [Contribution guide](contribute/#development-setup).
 
-{{< highlight class="runner-flink-local" >}}
-# Start a Flink job server on localhost:8099
-./gradlew :runners:flink:1.8:job-server:runShadow
+1. Clone the `beam` repository. It is recommended that you start from a stable
+   release branch rather than from master for both customizing the Dockerfile
+   and building image artifacts, and that you use the same version of the SDK
+   to run your pipeline with a custom container.
 
-# Run a pipeline on the Flink job server
-python -m apache_beam.examples.wordcount \
---input=/path/to/inputfile \
---output=/path/to/write/counts \
---runner=PortableRunner \
---job_endpoint=localhost:8099 \
---environment_config=path/to/container/image
-{{< /highlight >}}
+```
+export BEAM_SDK_VERSION="2.26.0"
 
-{{< highlight class="runner-spark-local" >}}
-# Start a Spark job server on localhost:8099
-./gradlew :runners:spark:job-server:runShadow
+git clone https://github.com/apache/beam.git
+git checkout origin/release-$BEAM_SDK_VERSION
+```
 
-# Run a pipeline on the Spark job server
-python -m apache_beam.examples.wordcount \
---input=/path/to/inputfile \
---output=path/to/write/counts \
---runner=PortableRunner \
---job_endpoint=localhost:8099 \
---environment_config=path/to/container/image
-{{< /highlight >}}
+3. Customize the `Dockerfile` for a given language. This file is typically in the `sdks/<language>/container` directory (e.g. the [Dockerfile for Python](https://github.com/apache/beam/blob/master/sdks/python/container/Dockerfile). If you're adding dependencies from [PyPI](https://pypi.org/), use [`base_image_requirements.txt`](https://github.com/apache/beam/blob/master/sdks/python/container/base_image_requirements.txt) instead.
 
-## Building container images
+3. Navigate to the root directory of the local copy of your Apache Beam.
```

Comment:
Does "Navigate to the root directory where you've installed your local copy of the Beam SDK." work?

Reply:
Tried to make it clearer (added some env var setting in instructions) - mostly worried installed might get confused for package installation (i.e. where pip installed Beam)